homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,137 @@
+"""Platform for light integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+
+# Import the device class from the component that you want to support
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zimi Light platform."""
+
+    api: ControlPoint = config_entry.runtime_data
+
+    entities = [ZimiLight(device, api) for device in api.lights]
+
+    async_add_entities(entities)
+
+
+class ZimiLight(LightEntity):
+    """Representation of a Zimi Light."""
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize an ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_should_poll = False
+        self._light = light
+        self._light.subscribe(self)
+        self._state = False
+        self._brightness = None
```

Comment:
why not just use the `_light` directly everywhere? You can maybe even get rid of the update method, if the object stays the same.

Reply:
Makes sense.    See https://github.com/home-assistant/core/pull/129876/commits/bc514d08fbad57233a3bed4b2a91c82d26f8fbae

Tested and working

Reply:
Wrong commit https://github.com/home-assistant/core/pull/129876/commits/ee32ea995a137c622266bc31751184313f704bcb